src/components/side-panel/layer-panel/layer-list.js

```diff
@@ -0,0 +1,174 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+
+import {arrayMove} from 'utils/data-utils';
+import styled from 'styled-components';
+import {createSelector} from 'reselect';
+import classnames from 'classnames';
+import {SortableContainer, SortableElement} from 'react-sortable-hoc';
+
+import LayerPanelFactory from './layer-panel';
+
+// make sure the element is always visible while is being dragged
+// item being dragged is appended in body, here to reset its global style
+const SortableStyledItem = styled.div`
+  z-index: ${props => props.theme.dropdownWrapperZ + 1};
+
+  &.sorting {
+    pointer-events: none;
+  }
+
+  &.sorting-layers .layer-panel__header {
+    background-color: ${props => props.theme.panelBackgroundHover};
+    font-family: ${props => props.theme.fontFamily};
+    font-weight: ${props => props.theme.fontWeight};
+    font-size: ${props => props.theme.fontSize};
+    line-height: ${props => props.theme.lineHeight};
+    *,
+    *:before,
+    *:after {
+      box-sizing: border-box;
+    }
+    .layer__drag-handle {
+      opacity: 1;
+      color: ${props => props.theme.textColorHl};
+    }
+  }
+`;
+
+LayerListFactory.deps = [LayerPanelFactory];
+
+function LayerListFactory(LayerPanel) {
+  // By wrapping layer panel using a sortable element we don't have to implement the drag and drop logic into the panel itself;
+  // Developers can provide any layer panel implementation and it will still be sortable
+  const SortableItem = SortableElement(({children, isSorting}) => {
+    return (
+      <SortableStyledItem className={classnames('sortable-layer-items', {sorting: isSorting})}>
+        {children}
+      </SortableStyledItem>
+    );
+  });
+
+  const WrappedSortableContainer = SortableContainer(({children}) => {
+    return <div>{children}</div>;
+  });
+
+  const LayerList = props => {
+    const {
+      layers,
+      datasets,
+      layerOrder,
+      uiStateActions,
+      visStateActions,
+      layerClasses,
+      isSortable
+    } = props;
+    const {toggleModal: openModal} = uiStateActions;
+    const [isSorting, setIsSorting] = useState(false);
+
+    const layerClassSelector = () => layerClasses;
+    const layerTypeOptionsSelector = createSelector(layerClassSelector, layerCls =>
```

Comment:
We are moving from reselect to react hooks. Here we should use `useMemo` instead of `createSelector`

Reply:
Since this is callable we need to use useCallback, not useMemo.